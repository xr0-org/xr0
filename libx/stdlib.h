#ifndef STDLIB_H
#define STDLIB_H

#include <stddef.h>

axiom void *
malloc(int size) ~ [
	setup: size = [0?];
	return .malloc(size);
];

axiom void
free(void *ptr) ~ [
	setup: ptr = malloc(1);
	.free(ptr);
];

axiom void
exit(int status);

#endif
