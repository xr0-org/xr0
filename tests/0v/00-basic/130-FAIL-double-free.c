#include <stdlib.h>

void
unit() 
{
	void *p;

	p = malloc(1);
	free(p);
	free(p);
}
