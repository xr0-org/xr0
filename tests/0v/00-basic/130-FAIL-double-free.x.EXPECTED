stdlib.h:14:24: precondition failure: argument of `ptr' must be lvalue (free)
	00-basic/130-FAIL-double-free.x:10:9 (unit)
